lock-api-objects/build.gradle

```diff
@@ -16,7 +16,7 @@ dependencies {
     implementation 'com.google.errorprone:error_prone_annotations'
     implementation 'com.google.guava:guava'
     implementation project(':commons-annotations')
-    testImplementation project(path: ':lock-api')
```

Comment:
this was an accidental change that was merged with #6647 . Just reverting this here and adding the option to use `KeyedStream` in tests